TypeCobol/AntlrGrammar/CobolCodeElements.g4

```diff
@@ -8312,6 +8313,22 @@ sqlGotoHostLabel: (SQL_GOTO | SQL_GO SQL_TO) ColonSeparator? hostLabel=UserDefin
 hostVariable: ColonSeparator mainVariable=UserDefinedWord ((indicator)? ColonSeparator indicatorVariable=UserDefinedWord)?;
 indicator: ({string.Equals(CurrentToken.Text, "INDICATOR", System.StringComparison.OrdinalIgnoreCase) }? KeywordINDICATOR=UserDefinedWord);
 
+restart: ({ string.Equals(CurrentToken.Text, "RESTART", System.StringComparison.OrdinalIgnoreCase) }? KeywordRESTART=UserDefinedWord);
+sqlIncrement: ({ string.Equals(CurrentToken.Text, "INCREMENT", System.StringComparison.OrdinalIgnoreCase) }? KeywordINCREMENT=UserDefinedWord);
+minvalue: ({ string.Equals(CurrentToken.Text, "MINVALUE", System.StringComparison.OrdinalIgnoreCase) }? KeywordMINVALUE=UserDefinedWord);
+maxvalue: ({ string.Equals(CurrentToken.Text, "MAXVALUE", System.StringComparison.OrdinalIgnoreCase) }? KeywordMAXVALUE=UserDefinedWord);
+cycle: ({ string.Equals(CurrentToken.Text, "CYCLE", System.StringComparison.OrdinalIgnoreCase) }? KeywordCYCLE=UserDefinedWord);
+cache: ({ string.Equals(CurrentToken.Text, "CACHE", System.StringComparison.OrdinalIgnoreCase) }? KeywordCACHE=UserDefinedWord);
+
+alterSequenceStatement: SQL_ALTER SQL_SEQUENCE (sequence_name=UserDefinedWord) alterSequenceClause alterSequenceClause*;
+alterSequenceClause: restartClause | incrementClause | minValueClause | maxValueClause | cacheClause | cycle | SQL_ORDER | noClauses;
+restartClause: restart ( SQL_WITH (numeric_constant=DecimalLiteral))? ;
+incrementClause: sqlIncrement SQL_BY (numeric_constant=DecimalLiteral);
+minValueClause: minvalue (numeric_constant=DecimalLiteral);
+maxValueClause: maxvalue (numeric_constant=DecimalLiteral);
+cacheClause: cache (numeric_constant=IntegerLiteral);
+noClauses: SQL_NO (minvalue | maxvalue | SQL_ORDER | cycle | cache);
```

Comment:
But this one dosn't word .If i write a sentence with "no maxValue" i get an error .The grammar is always waiting for a minValue after the NO because MinValue is the first one . That's why i changed that .

Reply:
Ok maybe ANTLR does not work properly with this grammar using a lot of contextual keyword. However you have to test for the presence or absence of 'NO' keyword so as to build more precise SyntaxProperty objects. A 'NO' keyword will translate into a non-null false SyntaxProperty<bool>.